lib/all_my_circuits/breaker.rb

```diff
@@ -14,10 +14,23 @@ def self.net_errors
       require "timeout"
       require "net/http"
 
-      [Timeout::Error, Errno::ECONNRESET, Errno::ECONNREFUSED, EOFError,
-       Net::HTTPBadResponse, Net::HTTPHeaderSyntaxError, Net::HTTPFatalError, Net::HTTPServerError]
+      [
+        EOFError,
+        Errno::ECONNREFUSED,
+        Errno::ECONNRESET,
+        Errno::ETIMEDOUT,
+        IOError,
+        Net::HTTPBadResponse,
+        Net::HTTPFatalError,
+        Net::HTTPHeaderSyntaxError,
+        Net::HTTPServerError,
```

Comment:
Net::HTTPServerError encapsulates most 5xx responses [source](https://github.com/ruby/ruby/blob/f845a9ef76c0195254ded79c85c24332534f4057/lib/net/http.rb#L379)
I think it should be replaced by it's subclasses:
```
  #   HTTPBadGateway::                      502
  #   HTTPServiceUnavailable::              503
  #   HTTPGatewayTimeOut::                  504
```